AdyenActions/Components/Voucher/VoucherViewControllerProvider.swift

```diff
@@ -32,44 +32,87 @@ internal final class VoucherViewControllerProvider: AnyVoucherViewControllerProv
     internal func provide(with action: VoucherAction) -> UIViewController {
         switch action {
         case let .dokuIndomaret(action):
-            let fields = createDokuVoucherFields(for: action)
-            return createGenericViewController(with: action, fields: fields)
+            return createGenericViewController(
+                with: createGenericView(with: action, fields: createDokuVoucherFields(for: action))
+            )
         case let .dokuAlfamart(action):
-            let fields = createDokuVoucherFields(for: action)
-            return createGenericViewController(with: action, fields: fields)
+            return createGenericViewController(
+                with: createGenericView(with: action, fields: createDokuVoucherFields(for: action))
+            )
         case let .econtextStores(action):
-            let fields = createEContextStoresVoucherFields(for: action)
-            return createGenericViewController(with: action, fields: fields)
+            return createGenericViewController(
```

Comment:
I don't think I like one return on 3 lines..
The only thing changing here is `fields`.
You can do:

```swift
internal func provide(with action: VoucherAction) -> UIViewController {
    let fields: [GenericVoucherView.VoucherField]
    switch action {
    case let .dokuIndomaret(action):
        fields = createDokuVoucherFields(for: action)
    case let .dokuAlfamart(action):
        fields = createDokuVoucherFields(for: action)
    case let .econtextStores(action):
        fields = createEContextStoresVoucherFields(for: action)
    case let .econtextATM(action):
        fields = createEContextATMVoucherFields(for: action)
    case let .boletoBancairoSantander(action):
        fields = createBoletoVoucherfields(for: action)
    }
    return createGenericViewController(with: createGenericView(with: action, fields: fields))
}
```

Reply:
unfortunately this won't work, since there are 2 different `createGenericView` methods. but `view` can be extracted in a similar manner..

Reply:
I updated my code snippet - doesn't that work either?

Reply:
it works. this is what i ended up with
``` swift
  internal func provide(with action: VoucherAction) -> UIViewController {
        let view: GenericVoucherView
        switch action {
        case let .dokuIndomaret(action):
            view = createGenericView(with: action, fields: createDokuVoucherFields(for: action))
        case let .dokuAlfamart(action):
            view = createGenericView(with: action, fields: createDokuVoucherFields(for: action))
        case let .econtextStores(action):
            view = createGenericView(with: action, fields: createEContextStoresVoucherFields(for: action))
        case let .econtextATM(action):
            view = createGenericView(with: action, fields: createEContextATMVoucherFields(for: action))
        case let .boletoBancairoSantander(action):
            view = createBoletoView(with: action)
        }
        
        return createGenericViewController(with: view)
    }
```

Reply:
👍